pkg/machine/compression/decompress.go

```diff
@@ -31,10 +32,7 @@ type decompressor interface {
 
 func Decompress(compressedVMFile *define.VMFile, decompressedFilePath string) error {
 	compressedFilePath := compressedVMFile.GetPath()
-	// Are we reading full image file?
-	// Only few bytes are read to detect
-	// the compression type
-	compressedFileContent, err := compressedVMFile.Read()
+	compressedFileContent, err := compressedVMFile.Peek(magicNumberMaxSize)
```

Comment:
Please rename the variable, and the parameter in `newDecompressor`, to not be misleading.

Reply:
Done.